com.minres.coredsl/src/com/minres/coredsl/util/ModelExtensions.xtend

```diff
@@ -59,8 +59,8 @@ abstract class ModelExtensions {
 		return null;
 	}
 
-	static def <T extends EObject> Iterator<T> descendantsOfType(EObject obj, Class<T> type) {
-		return obj.eAllContents.filter(type);
+	static def <T extends EObject> List<T> descendantsOfType(EObject obj, Class<T> type) {
```

Comment:
Is there a particular reason to switch to List?

Reply:
I'm not sure why this was ever `Iterator`. If anything, it should have been `Iterable`. We want to return something you can iterate over, not the iterator itself. Unfortunately `eAllContents` is an single-use iterator, so to make it iterable we need to collect it into a list.